analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -150,11 +150,11 @@ NumberConstraint From(BigInteger? newMin, BigInteger? newMax)
         {
             if (existingNumber is not null)
             {
-                if (!newMin.HasValue || (existingNumber.Min > newMin && IsFirstLoopVisit(visitCount)))
+                if ((newMin is null || (existingNumber.Min > newMin && IsFirstLoopVisit(visitCount))) && !(existingNumber.Min > newMax))
                 {
                     newMin = existingNumber.Min;
                 }
-                if (!newMax.HasValue || (existingNumber.Max < newMax && IsFirstLoopVisit(visitCount)))
+                if ((newMax is null || (existingNumber.Max < newMax && IsFirstLoopVisit(visitCount))) && !(existingNumber.Max < newMin))
                 {
                     newMax = existingNumber.Max;
                 }
```

Comment:
Alternatively, the local function could be implemented like this:
```NumberConstraint From(BigInteger? newMin, BigInteger? newMax)
{
    if (existingNumber is not null && IsFirstLoopVisit(visitCount))
    {
        if (newMin is null || existingNumber.Min > newMin)
        {
            newMin = existingNumber.Min;
        }
        if (newMax is null || existingNumber.Max < newMax)
        {
            newMax = existingNumber.Max;
        }
    }
    return NumberConstraint.From(newMin, newMax);
}
```
This would change some behavior, but you could argue that those changes make sense. I can demonstrate the difference for you if you want.

Reply:
That breaks `Loops_While_FixedCount` in undesired way